### PR TITLE
Vignettes

### LINKED ISSUE #49: Should we temporarily cache svg files?
Right now we store a single "temp.svg" file in the temporary directory. This is a part of the svg conversion process with `rsvg`. However, the ggplot production process often involves several iterations of tweaking plots before saving them, so perhaps it would make sense to store all of the svg files as \<uuid\>.svg in the temporary directory. Then, when we go to download a phylopic, we first check whether it is in the temporary directory first, instead of downloading and converting all over again. This should generally result in notable speed ups, especially when lots of the same phylopics are being used over and over. The temporary folder is cleared whenever R is closed, so this shouldn't result in any permanent storage usage.

### LINKED ISSUE #55: Vignette
Currently we have a few usage examples within our `README.md`. We should probably prepare a vignette which pulls this together. This would be beneficial for the following reasons:

- We will know if the examples need updating when there are changes to the package
- The examples are ported with the package
- The vignettes will be readily available via the pkgdown site

I suggest this should be done for v.1.1.0.

### PR DESCRIPTION
**Pull request by willgearty**
I've added three vignettes:

1. An introduction to the functionality of the package
2. Worked examples with ggplot2
3. Same worked examples with base R

Also, I discovered that there was a bug with `add_phylopic_base`, so I fixed that too.

Fixes #49 and fixes #55.

### CODE REVIEW

File: vignettes/b-advanced-ggplot.Rmd [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+```{r, echo = FALSE}
+penguin <- get_phylopic("86334821-42ec-4da1-bb9d-53f3d6941c77")
+```
+
+You may have noticed in the preview that the silhouette was a little slanted. Let's rotate it clockwise just a smidgen:
+
+```{r}
+# It's a little slanted, so let's rotate it a little bit
+penguin_rot <- rotate_phylopic(penguin, 15)
```
Comment (LewisAJones) [inline vignettes/b-advanced-ggplot.Rmd:57]: As it is an example, should we be explicit with the arguments?

File: vignettes/b-advanced-ggplot.Rmd [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+```{r}
+vertebrate.data$svg <- lapply(vertebrate.data$uuid, get_phylopic)
+```
+
+Now let's go ahead and plot our phylogeny with the [ggtree](https://www.bioconductor.org/packages/ggtree) package:
+
+```{r message = FALSE}
+library(ggtree)
```
Comment (LewisAJones) [inline vignettes/b-advanced-ggplot.Rmd:218]: Obviously this isn't currently available on CRAN for the most recent versions of R. Is this potentially problematic for the vignette building?

File: vignettes/b-advanced-ggplot.Rmd [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                     limits = c(-500, 0)) +
+  coord_geo_polar(dat = "periods") +
+  theme(line = element_line(linewidth = 1),
+        axis.text.r = element_text(size = 5, hjust = -0.5, vjust = -1.5))
+```
+
+Note that only a single size is specified and aspect ratio is always maintained, hence why the silhouettes all have the same height but different widths. Let's fix some of the silhouettes by rotating them 90 degrees:
+
+```{r}
+vertebrate.data$svg[[1]] <- rotate_phylopic(vertebrate.data$svg[[1]])
```
Comment (LewisAJones) [inline vignettes/b-advanced-ggplot.Rmd:253]: Again, as it is an example, should we be explicit with the arguments?

File: vignettes/c-advanced-base.Rmd [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+```{r, echo = FALSE}
+penguin <- get_phylopic("86334821-42ec-4da1-bb9d-53f3d6941c77")
+```
+
+You may have noticed in the preview that the silhouette was a little slanted. Let's rotate it clockwise just a smidgen:
+
+```{r}
+# It's a little slanted, so let's rotate it a little bit
+penguin_rot <- rotate_phylopic(penguin, 15)
```
Comment (LewisAJones) [inline vignettes/c-advanced-base.Rmd:55]: Same here, as it is an example, should we be explicit with the arguments?

File: R/add_phylopic_base.r [hunk 5]
```diff
@@ -145,28 +145,15 @@ add_phylopic_base <- function(img = NULL, name = NULL, uuid = NULL,
...
   if (is.null(x)) x <- mean(usr_x)
   if (is.null(y)) y <- mean(usr_y)
   if (is.null(ysize)) ysize <- abs(diff(usr_y))
 
-  # convert x, y, and ysize to percentages
-  x <- (x - xlims[1]) / diff(xlims)
-  y <- (y - ylims[1]) / diff(ylims)
-  ysize <- ysize / abs(diff(ylims))
+  # convert x, y, and ysize to normalized device coordinates
+  x <- grconvertX(x, to = "ndc")
```
Comment (LewisAJones) [inline R/add_phylopic_base.r:154]: I am glad this was an easy fix! Seems to work exactly as desired! Good job.

**Review by LewisAJones — APPROVED** [review]
Hi Will, 

This all looks good to me, great job on getting this finished and the updates to `add_phylopic_base` to handle multiple panels. 

I've left a few comments on the vignettes, but not much to say here as it largely echos the manuscript! My main worry is whether it's an issue that `ggtree` is not available via CRAN for the current R version? 

Lewis

File: vignettes/a-getting-started.Rmd [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+knitr::opts_chunk$set(echo = TRUE, include = TRUE, eval = TRUE,
+                      out.width = "100%", fig.width = 7.2, fig.height = 5,
+                      dpi = 200)
+```
+
+<div style="text-align: justify">
+
+# Introduction
+
+**rphylopic** is a package that allows users to easily fetch and visualize silhouettes of organisms from [PhyloPic](https://www.phylopic.org). The package allows users to add silhouettes as layers or as data points to both base R and `{ggplot2}` plots. Additional functionality allows users to pick between available silhouettes, transform them (e.g., rotate and recolor), and save image files. This vignette will give you an overview of the package and provide example usage.
```
[the remark this answers is not in the document]
Comment (willgearty) [reply vignettes/a-getting-started.Rmd:24]: This is a pkgdown standard for [linking to other packages](https://pkgdown.r-lib.org/articles/linking.html)

File: vignettes/a-getting-started.Rmd [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+To install the development version, first install the `{remotes}` package, and then use `install_github()` to install **rphylopic** directly from GitHub.
+
+```{r eval = FALSE}
+install.packages("remotes")
+remotes::install_github("palaeoverse-community/rphylopic")
+```
+
+You can now load **rphylopic** using the default `library()` function:
```
[the remark this answers is not in the document]
Comment (willgearty) [reply vignettes/a-getting-started.Rmd:41]: The other packages are all links, but it didn't make much sense to link to rphylopic, so I've used bold fontface.

File: DESCRIPTION [hunk 8]
```diff
@@ -40,9 +40,18 @@ Suggests:
     vdiffr (>= 1.0.0),
     knitr,
     rmarkdown,
-    covr
+    covr,
+    phytools,
+    palaeoverse,
+    ggtree,
```
[the remark this answers is not in the document]
Comment (willgearty) [reply DESCRIPTION:46]: ggtree is a bioconductor package. It's only in the `Suggests`, but I can check if we need to add anything to the description file to make sure it works

**Pull request merged**